src/connect/connect.cpp

```diff
@@ -312,11 +319,21 @@ optional<OnlineStatus> connect::communicate(CachedFactory &conn_factory) {
         }
     });
 
-    printer.renew();
-
     HttpClient http(conn_factory);
 
-    BasicRequest request(printer, config, action);
+    uint32_t now = ticks_ms();
+    // Underflow should naturally work
+    if (now - last_full_telemetry >= FULL_TELEMETRY_EVERY) {
+        // The server wants to get a full telemetry from time to time, despite
+        // it not being changed. Some caching reasons/recovery/whatever?
+        //
+        // If we didn't send a new telemetry for too long, reset the
+        // fingerprint, which'll trigger the resend.
+        telemetry_changes.mark_dirty();
+    }
+    const bool is_full_telemetry = holds_alternative<SendTelemetry>(action) && !get<SendTelemetry>(action).empty;
```

Comment:
minor nitpick: Would maybe be more readable if moved closer to where it is actually used.

Reply:
I moved it a bit, but it's only a few lines.